### ISSUE TITLE
alembic does not autogenerate engine for clickhouse table

### ISSUE THREAD

**Issue by solidguy7** [xzkostyan/clickhouse-sqlalchemy#299]
**Describe the bug**
I have prepared some table in order to test alembic autogeneration:

```
from clickhouse_sqlalchemy import types, engines
from sqlalchemy.ext.asyncio import create_async_engine
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column

from dwh.main.config import settings

engine = create_async_engine(url=settings.DB_URL)


class Base(DeclarativeBase):
    pass


class Incident(Base):
    incident_id: Mapped[int] = mapped_column(types.Int8, primary_key=True)

    __tablename__ = "incidents"
    __table_args__ = (
        engines.MergeTree(order_by=incident_id),
    )
```
My DB_URL is `DB_URL=clickhouse+asynch://default:default@localhost:9000/default`

When I enter `alembic revision --autogenerate -m 'init' `, I will get the following migration:

```
from typing import Sequence, Union

from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision: str = "e7cfe512c3a1"
down_revision: Union[str, None] = None
branch_labels: Union[str, Sequence[str], None] = None
depends_on: Union[str, Sequence[str], None] = None


def upgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table(
        "incidents",
        sa.Column(
            "incident_id", clickhouse_sqlalchemy.types.common.Int8(), nullable=False
        ),
        sa.PrimaryKeyConstraint("incident_id"),
    )
    # ### end Alembic commands ###


def downgrade() -> None:
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table("incidents")
    # ### end Alembic commands ###
```

my env.py:

```
import asyncio
from logging.config import fileConfig

from dwh.core.database import Base
from dwh.main.config import settings
from sqlalchemy import pool
from sqlalchemy.engine import Connection
from sqlalchemy.ext.asyncio import async_engine_from_config
from clickhouse_sqlalchemy import engines, types

from alembic.ddl import impl
from alembic import context

# this is the Alembic Config object, which provides
# access to the values within the .ini file in use.

config = context.config
config.set_main_option("sqlalchemy.url", settings.DB_URL)

# Interpret the config file for Python logging.
# This line sets up loggers basically.
if config.config_file_name is not None:
    fileConfig(config.config_file_name)

# add your model's MetaData object here
# for 'autogenerate' support
# from myapp import mymodel
# target_metadata = mymodel.Base.metadata
target_metadata = Base.metadata

# other values from the config, defined by the needs of env.py,
# can be acquired:
# my_important_option = config.get_main_option("my_important_option")
# ... etc.


class CustomDialectImpl(impl.DefaultImpl):
    __dialect__ = "clickhouse"
    transactional_ddl = False


def run_migrations_offline() -> None:
    """Run migrations in 'offline' mode.

    This configures the context with just a URL
    and not an Engine, though an Engine is acceptable
    here as well.  By skipping the Engine creation
    we don't even need a DBAPI to be available.

    Calls to context.execute() here emit the given string to the
    script output.

    """
    url = config.get_main_option("sqlalchemy.url")
    context.configure(
        url=url,
        target_metadata=target_metadata,
        literal_binds=True,
        dialect_opts={"paramstyle": "named"},
    )

    with context.begin_transaction():
        context.run_migrations()


def do_run_migrations(connection: Connection) -> None:
    context.configure(connection=connection, target_metadata=target_metadata)

    with context.begin_transaction():
        context.run_migrations()


async def run_async_migrations() -> None:
    """In this scenario we need to create an Engine
    and associate a connection with the context.

    """

    connectable = async_engine_from_config(
        config.get_section(config.config_ini_section, {}),
        prefix="sqlalchemy.",
        poolclass=pool.NullPool,
    )

    async with connectable.connect() as connection:
        await connection.run_sync(do_run_migrations)

    await connectable.dispose()


def run_migrations_online() -> None:
    """Run migrations in 'online' mode."""

    asyncio.run(run_async_migrations())


if context.is_offline_mode():
    run_migrations_offline()
else:
    run_migrations_online()
```

**Expected behavior**
I expect that an engine will be generated

**Versions**
- clickhouse-sqlalchemy 0.3.0
- alembic 1.13.1.
- Python 3.12.1.

**Comment by piroinno** [comment]
Looks like this is a known issue (https://clickhouse-sqlalchemy.readthedocs.io/en/latest/migrations.html#limitations).

**Comment by solidguy7** [comment]
Very sadly that there is no way to autogenerate migrations(
I have to manually generate each migration, even if I wanna add some column to my table

**Comment by piroinno** [comment]
> Very sadly that there is no way to autogenerate migrations( I have to manually generate each migration, even if I wanna add some column to my table

This issue has been fixed with the most recent version 0.3.1. auto generations now works. But the engines issue still persists

**Comment by solidguy7** [comment]
I`ve already solved this issue by using django-clickhouse-backend. There are no any problems with autogenerating and engines. I consider that it`s a good idea because I need only orm and migrations
